### ISSUE TITLE
False negative when type-checking calls to methods of generic superclass

### ISSUE THREAD

**Issue by itaisteinherz** [python/mypy#11107]
**Bug Report**

While type-checking my code, I noticed a bug which occurs if you type-check generic classes with mypy, where one generic class inherits from another. When creating instances of the subclass, and calling one of the base class's methods on an instance, mypy entirely ignores the return type, and allows you to do anything with the returned object.

_(Note: I'm not sure I'm using the correct terminology here, and would appreciate it if you could call out any inaccuracies.)_

**To Reproduce**

1. Save the following code in `reproduction.py`:

```python
from typing import Generic, TypeVar 


I = TypeVar('I')


class A(Generic[I]):
	def get_i(self) -> I: pass 


class B(A, Generic[I]):
	pass


b = B[int]()
b.get_i().upper()  # invalid, but doesn't get reported
```

2. Type-check `reproduction.py`.

**Expected Behavior**

I expected mypy to report an error for the last line, as it contains a type error.

**Actual Behavior**

mypy doesn't report any errors for it.

Note that the expression can be replaced with other equal cases, yielding the same result, e.g.

```python
b.get_i().some_nonexistent_method()
```

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.9.1 (I also experienced this on Python 3.9.7)
- Operating system and version: Windows 10 (Version 20H2 (OS Build 19042.804))

**Comment by erictraut** [comment]
You need to declare `B` like this:
```python
class B(A[I]):
    pass
```

The way you've declared it is treated as:
```python
class B(A[Any], Generic[I]):
    pass
```

**Comment by itaisteinherz** [comment]
@erictraut Thanks for the quick reply!

That was indeed my mistake, the bug isn't in mypy 😅
